tools/android/packaging/xbmc/src/XBMCJsonRPC.java.in

```diff
@@ -1181,108 +1183,139 @@ public class XBMCJsonRPC
+    library_items.put("movies", 0);
+    library_items.put("tvshows", 0);
+    library_items.put("albums", 0);
+
     JsonObject rep = request_object(RECOMMENDATION_MOVIES_JSON);
+    JsonArray movies = null;
     if (rep != null && rep.has("result"))
     {
       try
       {
         JsonObject results = rep.getAsJsonObject("result");
         if (results.has("movies"))
         {
-          JsonArray movies = results.get("movies").getAsJsonArray();
-
-          int count = 0;
-          for (int i = 0; i < movies.size() && count < MAX_RECOMMENDATIONS; ++i)
-          {
-            try
-            {
-              JsonObject details = movies.get(i).getAsJsonObject();
-              Movie med = createMovieFromJson(details);
-              if (med != null)
-              {
-                medias.add(med);
-                ++count;
-              }
-            }
-            catch (Exception e)
-            {
-              continue;
-            }
-          }
+          movies = results.get("movies").getAsJsonArray();
+          library_items.put("movies", movies.size());
         }
-      } catch (Exception e)
-      {
-        e.printStackTrace();
+      } catch (Exception e) {
       }
     }
 
     rep = request_object(RECOMMENDATIONS_SHOWS_JSON);
+    JsonArray tvshows = null;
     if (rep != null && rep.has("result"))
     {
       try
       {
         JsonObject results = rep.getAsJsonObject("result");
         if (results.has("tvshows"))
         {
-          JsonArray tvshows = results.get("tvshows").getAsJsonArray();
-
-          int count = 0;
-          for (int i = 0; i < tvshows.size() && count < MAX_RECOMMENDATIONS; ++i)
-          {
-            try
-            {
-              JsonObject tvshow = tvshows.get(i).getAsJsonObject();
-              TVShow med = createTVShowFromJson(tvshow);
-              if (med != null)
-              {
-                medias.add(med);
-                ++count;
-              }
-            }
-            catch (Exception e)
-            {
-              continue;
-            }
-          }
+          tvshows = results.get("tvshows").getAsJsonArray();
+          library_items.put("tvshows", tvshows.size());
         }
-      } catch (Exception e)
-      {
-        e.printStackTrace();
+      } catch (Exception e) {
       }
     }
 
     rep = request_object(RECOMMENDATIONS_ALBUMS_JSON);
+    JsonArray albums = null;
     if (rep != null && rep.has("result"))
     {
       try
       {
         JsonObject results = rep.getAsJsonObject("result");
         if (results.has("albums"))
         {
-          JsonArray albums = results.get("albums").getAsJsonArray();
+          albums = results.get("albums").getAsJsonArray();
+          library_items.put("albums", albums.size());
+        }
+      } catch (Exception e) {
+      }
+    }
+
+    int zeros = 0;
+    for (Integer value : library_items.values()) {
+      if (value == 0) {
+        ++zeros;
+      }
```

Comment:
You don't need `libraray_item`and also not this loop. Just define `zeros` where you now define `library_items` and increase `zeros`directly after the three `results.get("foo").getAsJsonArray();` in case `foo.size()` is zero

Reply:
> You don't need `libraray_item`and also not this loop. Just define `zeros` where you now define `library_items` and increase `zeros`directly after the three `results.get("foo").getAsJsonArray();` calls, in case `foo.size()` is zero.

You're right, I added the hashmap with an initial idea, but in the end it wasn't necessary, thanks.

> And I would invert the logic, not counting `zeros` but number of categories with data. Feels more natural and code is better readable.

Yes, feels more natural, change it.

> But you also can completely eliminate `zeros` or `catagories with data`. After fetching all data via the three `results.get("foo").getAsJsonArray();` calls calculate `MAX_RECOMMENDATIONS` based on for example `movies` not being null and `movies` size greater 0... you get the idea.

I tried it but it looks like an if/else with a lot of conditions, not so readable, it seems simpler as it is now.